### PR TITLE
ARGO-4178 Optimize xml parsing with lxml

### PR DESCRIPTION
**Pull request by deksa89**

### CODE REVIEW

**Comment by deksa89** [comment]
merge this branch after refactor-plainjson-write branch

File: modules/io/jsonwrite.py [hunk 1]
```diff
@@ -1,9 +1,6 @@
-#!/usr/bin/env python3
-
 import json
 
-
-class JsonWriter:
+class JsonWriter():
```
Comment (vrdel) [inline modules/io/jsonwrite.py:3]: Derive here from "object", so `class JsonWriter(object)`. `object` is base for all new classes - starting from Python 2. Check for example https://stackoverflow.com/questions/4015417/why-do-python-classes-inherit-object

File: modules/parse/base.py [hunk 2]
```diff
@@ -14,95 +14,91 @@ class ParseHelpers(object):
...
+                    key = ext_node.text
+                if ext_node.tag == 'VALUE':
+                    value = ext_node.text
+                if key and value:
+                    extensions_dict.update({key: value})
 
         return extensions_dict
 
-    def parse_url_endpoints(self, endpoints_node):
+    def parse_url_endpoints_lxml(self, endpoints_node):
```
Comment (vrdel) [inline modules/parse/base.py:32]: Preserve old name of methods without `_lxml` suffix.

File: tests/test_servicetypefeed.py [hunk 3]
```diff
@@ -25,109 +25,110 @@ def setUp(self):
     def test_WebApiFeedParse(self):
         service_types = self.services_webapi.get_data()
         self.assertEqual(service_types,
```
Comment (vrdel) [inline tests/test_servicetypefeed.py:27]: Fix indents here.

**Review by vrdel — CHANGES_REQUESTED** [review]
Next to single comments provided in PR, I'm also seeing:
* `parse_xml()` in `base.py` still rely on old library
* `ExpatError` handling that is leftover from previous library
* imports of old library like `import xml.dom...` and `from xml.parsers...`

Take care of these also.

**Review by vrdel — CHANGES_REQUESTED** [review]
You need to parse `NOTIFICATIONS` field for service endpoints and service groups as well. Currently, you're doing it only for sites. Take a look at my branch `notif-flag-honor` and check there at `gocdb_topology.py` how I'm doing it for the rest of topology components.

Also, there's a new config option that you need to place in `customer.conf` file to test it - example:
```
[CUSTOMER_SDC]
....
HonorNotificationFlag = True
....
```

**Comment by vrdel** [comment]
Merging and fixing in later PRs.

**Pull request merged**